x/incentive/types/genesis.go

```diff
@@ -12,9 +15,35 @@ type GenesisClaimPeriodID struct {
 	ID    uint64 `json:"id" yaml:"id"`
 }
 
+// Validate performs a basic check of a GenesisClaimPeriodID fields.
+func (gcp GenesisClaimPeriodID) Validate() error {
+	if gcp.ID == 0 {
+		return errors.New("genesis claim period id cannot be 0")
+	}
+	return sdk.ValidateDenom(gcp.Denom)
+}
+
 // GenesisClaimPeriodIDs array of GenesisClaimPeriodID
 type GenesisClaimPeriodIDs []GenesisClaimPeriodID
 
+// Validate checks if all the GenesisClaimPeriodIDs are valid and there are no duplicated
+// entries.
+func (gcps GenesisClaimPeriodIDs) Validate() error {
+	seenIDS := make(map[string]bool)
+	for _, gcp := range gcps {
+		if seenIDS[gcp.Denom+string(gcp.ID)] {
```

Comment:
nit: `gcp.Denom+string(gcp.ID)` is used twice, so I'd recommend storing this into a variable.